exercise/serializer.py

```diff
@@ -114,6 +116,11 @@ def get_to_account(self, email):
     def get_reveal_time(self, email):
         return email.reveal_time
 
+    def get_thread_properties(self, email):
```

Comment:
as discussed, once exercise is passed into context it's easy to pass exercise to `exercize_specific_properties`
```python
def get_thread_properties(self, email):
    return ExerciseEmailPropertiesSerializer(
        email.exercise_specific_properties(exercise=self.context.get("exercise"))
    ).data
```

Reply:
You can add exercise to the context at `ExerciseSerializer`.
```python
def get_threads(self, exercise):
    queryset = exercise.emails.all().filter(pk=F("belongs_to"))
    return ThreadSerializer(queryset, many=True, context={'exercise': exercise}).data
```